src/modules/auth/auth.service.ts

```diff
@@ -30,4 +45,31 @@ export class AuthService {
 			refreshToken: this.tokenService.generateRefreshToken(user)
 		};
 	}
+
+	async handleGoogleAuth({ accessToken }: GoogleLoginDto) {
+		try {
+			const googleUserInfo: GoogleUserInfo = await this.fetchGoogleUserProfile(accessToken);
+			const existingUser = await this.userService.handleGetUserByEmail(googleUserInfo.email);
+			// If user is not registered
+			if (!existingUser) return this.userService.handleCreateGoogleUser(googleUserInfo);
+			// If user already exist with such email
+			else return this.login(existingUser);
```

Comment:
No need for else here. You already early exit the function if the first condition is true. Also. personal note: using curly brackets is always better for code redability.
Compare:
`if (!existingUser) return this.userService.handleCreateGoogleUser(googleUserInfo);
			// If user already exist with such email
			else return this.login(existingUser);`
			
and

`if (!existingUser) {
    return this.userService.handleCreateGoogleUser(googleUserInfo);
}
// If user already exist with such email
return this.login(existingUser);
`